tools/pylib/_boutcore_build/boutcore.pyx.in

```diff
@@ -1118,17 +1175,27 @@ def init(args=[]):
         raise RuntimeError("Failed to initialise the BOUT++ Library (Error code %d)"%ret)
     else:
         _isInit=True
-    #atexit.register(finalise)
+    atexit.register(finalise)
 
 def finalise():
     """
     Finalize BOUT++ and also MPI.
     After this most objects and functions are not valid anymore.
     """
-    checkInit()
-    BoutFinalise()
     global _isInit
-    _isInit=False
+    import gc
+    ls = gc.get_objects()
+    im=(Coordinates, Field2D, Field3D, Laplacian, Mesh, Options,
```

Comment:
Please could you give these more descriptive names? It's hard to work out what's going on

Reply:
I also added some documentation, I hope that helps